settings/src/main/java/org/odk/collect/settings/enums/StringIdEnumUtils.kt

```diff
@@ -0,0 +1,29 @@
+package org.odk.collect.settings.enums
+
+import android.content.Context
+import org.odk.collect.settings.keys.ProjectKeys
+import org.odk.collect.shared.settings.Settings
+
+object StringIdEnumUtils {
+
+    @JvmStatic
+    fun Settings.getFormUpdateMode(context: Context): FormUpdateMode {
+        val setting = this.getString(ProjectKeys.KEY_FORM_UPDATE_MODE)
+        return parse(context, setting)
+    }
+
+    @JvmStatic
+    fun Settings.getAutoSend(context: Context): AutoSend {
+        val setting = this.getString(ProjectKeys.KEY_AUTOSEND)
+        return parse(context, setting)
+    }
+
+    private inline fun <reified T> parse(
```

Comment:
This is a pretty intense method signature, but it prevents us needing to create a `parse` for each of these enum settings (which I can see us adding more of).

We could actually also have a `Settings.getEnum` extension that would remove the need for the `getFormUpdateMode`/`getAutoSend` methods, but this ends up not being callable from Java sadly (due to the use of `inline`/`reified`).